lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,327 @@
+                Output cube containing Nowcast lightning probability.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over('time'):
+            thispoint = cube_slice.coord('time').points[0]
+            thistime = iris_time_to_datetime(
+                cube_slice.coord('time').copy())[0]
+            this_ltng = ltng_cube.extract(iris.Constraint(time=thistime))
+            fg_time = iris_time_to_datetime(
+                fg_cube.coord('time').copy())[
+                fg_cube.coord('time').nearest_neighbour_index(thispoint)]
+            this_fg = fg_cube.extract(iris.Constraint(time=fg_time))
+            err_string = "No matching {} cube for {}"
+            assert isinstance(this_ltng,
+                              iris.cube.Cube), err_string.format("lightning",
+                                                                 thistime)
+            assert isinstance(cube_slice,
+                              iris.cube.Cube), err_string.format("output",
+                                                                 thistime)
+            assert isinstance(this_fg,
+                              iris.cube.Cube), err_string.format("first-guess",
+                                                                 thistime)
+            cube_slice.data = this_fg.data
+            cube_slice.coord('forecast_period').convert_units('minutes')
+            fcmins = cube_slice.coord('forecast_period').points[0]
+
+            # Increase prob(lightning) to Risk 2 (pl_dict[2]) when within
+            #   lightning halo (lrt_lev2; 50km of an observed ATDNet strike):
+            cube_slice.data = np.where(
+                np.logical_and(this_ltng.data >= self.lrt_lev2,
+                               cube_slice.data < self.pl_dict[2]),
+                self.pl_dict[2], cube_slice.data)
+            lratethresh = self.lrt_lev1(fcmins)
+            if self.debug:
+                print('LRate threshold is {} strikes per minute'.format(
+                    lratethresh))
+
+            # Increase prob(lightning) to Risk 1 (pl_dict[1]) when within
+            #   lightning storm (lrt_lev1; ~5km of an observed ATDNet strike):
+            cube_slice.data = np.where(
+                np.logical_and(this_ltng.data >= lratethresh,
+                               cube_slice.data < self.pl_dict[1]),
+                self.pl_dict[1], cube_slice.data)
+
+            new_cube_list.append(cube_slice)
+
+        merged_cube = new_cube_list.merge_cube()
+        merged_cube = check_cube_coordinates(
+            cube, merged_cube)
+
+        # Apply precipitation adjustments.
+        merged_cube = self.precip_plugin.process(merged_cube, precip_cube)
+
+        # If we have VII data, increase prob(lightning) accordingly.
+        if vii_cube:
+            merged_cube = self.ice_plugin.process(merged_cube, vii_cube)
+        return merged_cube
+
+    def process(self, cubelist):
```

Comment:
I think this would be clearer with a list of named variables, ie:
```
def process(self, lightning_rate, prob_lightning_cube, prob_precip_cube, vii_cube=None)
```